### PR TITLE
feat(frontend):Updated CCN tab

### PR DESCRIPTION
**Pull request by nkrmr**
- Updated ccn tab on "DureePreavisDemission" tool
- Updated "DureePreavisDemission" test

### CODE REVIEW

**Comment by lionelB** [comment]
~~don't forget to fix codecov before merging :)~~  wrong PR

File: packages/code-du-travail-data/dataset/preavis-demission/index.js [hunk 1]
```diff
@@ -41,7 +42,7 @@ function getHeaders(row) {
 async function getData() {
   const { cells } = await getCells({
     key: SPREADSHEET_KEY,
-    worksheet: 2
+    worksheet: 5
```
Comment (lionelB) [inline packages/code-du-travail-data/dataset/preavis-demission/index.js]: this should be a constant

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+// store selected convention in localStorage
+const useConventionState = createPersistedState("convention");
+
+function StepInfoCCn({ form }) {
+  const [ccInfo, setCcInfo] = useConventionState({});
+  useEffect(() => {
+    form.batch(() => {
+      [
+        "contract",
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js]: these key are defined in the PrimePrecarite tools. We can loop over `getState().values` key  to achieve the same thing and make it more general. That way we extract this step to `commons`  and use it in each tool.

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        }}
+      />
+    </>
+  );
+}
+
+StepInfoCCn.validate = values => {
+  const errors = {};
+  if (Object.keys(values).length === 0) {
+    errors.ccn = "Quelle est votre convention collective ?";
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js]: `Veuillez renseigner votre convention collective ?`

Comment (lionelB) [reply packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js]: I  don't see how can i have this error message

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  } else if (values.ccn && isNotYetProcessed(values.ccn.num)) {
+    errors.ccn =
+      "Le code du travail ne prévoit pas une durée précise du préavis de démission. Il prévoit qu'un texte du niveau légal conventionnel, un accord d'entreprise voire un usage en prévoit les durées et modalités. ";
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js]: the error message not reflect the fact that the ccn is not included in our tools

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/Result.js [hunk 5]
```diff
@@ -3,16 +3,26 @@ import PropTypes from "prop-types";
 import { Toast } from "@socialgouv/react-ui";
 
 import { SectionTitle } from "../../common/stepStyles";
-import { ccCriterionName, filterSituations, recapSituation } from "./situation";
+import {
+  filterSituations,
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/Result.js:10]: filterSituation / getSituationFor should be in utils too

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/situation.js [hunk 6]
```diff
@@ -168,3 +82,14 @@ export function recapSituation(criteria) {
       null
     );
 }
+
+export function getSituationsFor(obj) {
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/situation.js]: can you make this function generic getSituationFor(data, values) => situations[]

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/situation.js [hunk 7]
```diff
@@ -168,3 +82,14 @@ export function recapSituation(criteria) {
...
+  return data.filter(situation =>
+    Object.entries(obj).every(([key, value]) => situation[key] === value)
+  );
+}
+
+export const isNotYetProcessed = idcc => {
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/situation.js]: same here this function could be more generic and move to utils (and probably used in IndemnitePrecarité)

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js (hunk 4 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js]: C'est le message que m'a demandé de rajouter Jean René si l'ont ne match pas de résultat

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js (hunk 3 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js]: Done

File: packages/code-du-travail-data/dataset/preavis-demission/index.js (hunk 1 above)
Comment (nkrmr) [reply packages/code-du-travail-data/dataset/preavis-demission/index.js]: Done

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js (hunk 2 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js]: OK i'll do that

File: packages/code-du-travail-frontend/src/outils/utils.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
+const createValuesMatcher = values => item => {
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/outils/utils.js]: I would prefer named the file situation.{utils|lib}.js or just situation.js  and remane local situations.js to  another name (it only contains mapping between string / constants) for display purpose

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/Result.js (hunk 5 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/Result.js:10]: They already are on utils but exported within situation

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/situation.js (hunk 6 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/situation.js]: OK

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/situation.js (hunk 7 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/situation.js]: OK

File: packages/code-du-travail-frontend/src/outils/utils.js (hunk 8 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/utils.js]: OK i understand and i'm agree with you

Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/utils.js]: I think it's more clear to keep situations.js on local and rename utils.js to situations.utils.js

Comment (lionelB) [reply packages/code-du-travail-frontend/src/outils/utils.js]: ok with that but all the situation logic should go into situation.utils

Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/utils.js]: Yes

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/situation.js (hunk 6 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/situation.js]: Done

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/situation.js (hunk 7 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/situation.js]: Done

File: packages/code-du-travail-frontend/src/outils/utils.js (hunk 8 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/utils.js]: Done

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+`;
+
+const { spacing } = theme;
+const P = styled.p`
+  font-style: italic;
+`;
+const StyledToast = styled(Toast)`
+  width: 100%;
+  margin-top: ${spacing.interComponent};
+`;
```
Comment (UnbearableBear) [inline packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js]: Est-ce qu'on ne pourrait pas généraliser / factoriser cette step de demande de CCN ?

Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js]: Si je suis en train de m'en occuper

**Review by UnbearableBear — APPROVED** [review]
Il te reste un conflit à régler mais 👍

File: packages/code-du-travail-frontend/src/outils/common/InfosCCn.js [hunk 10]
```diff
@@ -15,30 +16,17 @@ function StepInfoCCn({ form }) {
...
+      Object.keys(form.getState().values)
+        .filter(key => key !== CCN)
+        .forEach(key => form.change(key, undefined));
+      form.change(CCN, ccInfo.convention);
     });
   }, [ccInfo, form]);
   return (
     <>
       <Field
         name="ccn"
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/outils/common/InfosCCn.js]: if we use a constant for the CCN key name, we should use it here too (and probably in other place)

File: packages/code-du-travail-frontend/src/outils/common/InfosCCn.js [hunk 11]
```diff
@@ -4,9 +4,10 @@ import styled from "styled-components";
 import createPersistedState from "use-persisted-state";
 import { Button, Toast, theme } from "@socialgouv/react-ui";
 
-import { QuestionLabel } from "../../common/stepStyles";
-import Search from "../../../conventions/Search/Form";
-import { isNotYetProcessed } from "./situation";
+import { QuestionLabel } from "./stepStyles";
+import Search from "../../conventions/Search/Form";
+
+const CCN = "ccn";
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/outils/common/InfosCCn.js]: you can export it and use it elsewhere

**Review by lionelB — COMMENTED** [review]
@nkrmr I'm not sure but from the code below, the step InfoCCn will not show our original warning message (ccn is not yet  handled)  without blocking the progression in the wizard.

File: packages/code-du-travail-frontend/src/outils/common/InfosCCn.js (hunk 10 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/common/InfosCCn.js]: Done

File: packages/code-du-travail-frontend/src/outils/common/InfosCCn.js (hunk 11 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/common/InfosCCn.js]: Done

**Comment by nkrmr** [comment]
> @nkrmr I'm not sure but from the code below, the step InfoCCn will not show our original warning message (ccn is not yet handled) without blocking the progression in the wizard.

@lionelB You talk about the code on Result.js?

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/__tests__/situation.test.js [hunk 12]
```diff
@@ -1,114 +1,97 @@
 import {
+  getSituationsFor,
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/__tests__/situation.test.js]: you don't need to keep tests other than `recapSituation()` in this file.

File: packages/code-du-travail-frontend/src/outils/IndemnitePrecarite/steps/__tests__/InfoCCn.test.js [hunk 13]
```diff
@@ -1,6 +1,6 @@
 import React from "react";
 import { render } from "@testing-library/react";
-import { StepInfoCCn } from "../InfosCCn";
+import { StepInfoCCn } from "../../../common/InfosCCn";
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/outils/IndemnitePrecarite/steps/__tests__/InfoCCn.test.js]: you can move this tests file to common

File: packages/code-du-travail-frontend/src/outils/IndemnitePrecarite/steps/__tests__/situations.test.js [hunk 14]
```diff
@@ -3,10 +3,9 @@ import {
   filterSituations,
   getNextQuestionKey,
   getOptions,
-  getPastQuestions,
-  validateSituation,
-  isNotYetProcessed
-} from "../situation";
+  getPastQuestions
+} from "../../../common/situations.utils";
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/outils/IndemnitePrecarite/steps/__tests__/situations.test.js]: can you keep only tests that are related to `situation.js` and move other tests to `situations.utils.test.js` ?

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/__tests__/situation.test.js (hunk 12 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/__tests__/situation.test.js]: OK

File: packages/code-du-travail-frontend/src/outils/IndemnitePrecarite/steps/__tests__/InfoCCn.test.js (hunk 13 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/IndemnitePrecarite/steps/__tests__/InfoCCn.test.js]: Done

File: packages/code-du-travail-frontend/src/outils/IndemnitePrecarite/steps/__tests__/situations.test.js (hunk 14 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/IndemnitePrecarite/steps/__tests__/situations.test.js]: Done

File: packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/__tests__/situation.test.js (hunk 12 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/__tests__/situation.test.js]: Done

**Review by lionelB — APPROVED** [review]

**Pull request merged**